### PR TITLE
feat(Core/Common): move EventMap to separated files. Add support chrono

### PR DESCRIPTION
**Pull request by Winfidonarleyan**
## Changes Proposed:
- Move EventMap to separated files
- Add support `std::chrono::duration`

## Issues Addressed:
- Part #6490 

## Tests Performed:
- Locally

### CODE REVIEW

File: src/common/Utilities/EventMap.h [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+
+    /**
+    * @name Update
+    * @brief Updates the timer of the event map.
+    * @param time Value in ms to be added to time.
+    */
+    void Update(Milliseconds time)
+    {
+        _time += time.count();
+    }
```
Comment (Kitzunu) [inline src/common/Utilities/EventMap.h:67]: What does this output? I have read somewhere that chrono to uint32 can be weird.
Just want to make sure we dont break something

Comment (Winfidonarleyan) [reply src/common/Utilities/EventMap.h:67]: You about `.count()` ?

Comment (Kitzunu) [reply src/common/Utilities/EventMap.h:67]: yea

Comment (Winfidonarleyan) [reply src/common/Utilities/EventMap.h:67]: for `Milliseconds` return `uint64`

Comment (Kitzunu) [reply src/common/Utilities/EventMap.h:67]: great

Comment (Winfidonarleyan) [reply src/common/Utilities/EventMap.h:67]: i added `static_cast`

**Pull request merged**